### PR TITLE
Small changes to help implement routing

### PR DESCRIPTION
**Pull request by SamGodwin2**
### What is the context of this PR?
Updates to the schema based on the work coming out of implementing routing on the api.

### How to review 
Changes make sense, and are logical.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -395,43 +402,67 @@ input DeleteRoutingRuleSetInput {
...
 
+input UndeleteRoutingRuleInput {
+  id: ID!
+}
+
 input CreateRoutingConditionInput {
   comparator: RoutingComparator!
   answerId: ID!
-  routingValue: RoutingConditionValueInput!
+  parentRoutingRule: ID!
```
Comment (samiwel) [inline index.js]: same, `routingRuleId`?

File: index.js [hunk 2]
```diff
@@ -395,43 +402,67 @@ input DeleteRoutingRuleSetInput {
...
 }
 
-input RoutingConditionValueInput {
+input UndeleteRoutingConditionInput {
+  id: ID!
+}
+
+input CreateRoutingConditionValueInput {
   idArrayValue: [ID]
+  parentCondition: ID!
```
Comment (samiwel) [inline index.js]: `routingConditionId`?

File: index.js [hunk 3]
```diff
@@ -395,43 +399,53 @@ input DeleteRoutingRuleSetInput {
...
 
-input RoutingConditionValueInput {
-  idArrayValue: [ID]
+input CreateRoutingValueWithOptionIdInput {
+  value: ID!
+  condition: ID!
+}
+
+input DeleteRoutingConditionValueInput {
+  id: [ID]!
```
Comment (WickyNilliams) [inline index.js]: We don't need this, or the associated mutation

File: index.js [hunk 4]
```diff
@@ -395,43 +399,53 @@ input DeleteRoutingRuleSetInput {
...
 }
 
 input DeleteRoutingConditionInput {
   id: ID!
 }
 
-input RoutingConditionValueInput {
-  idArrayValue: [ID]
+input CreateRoutingValueWithOptionIdInput {
+  value: ID!
```
Comment (WickyNilliams) [inline index.js]: rename `ToggleConditionOptionInput`, and give mutation similar name

File: index.js [hunk 5]
```diff
@@ -395,43 +399,53 @@ input DeleteRoutingRuleSetInput {
...
+
 input CreateRoutingConditionInput {
   comparator: RoutingComparator!
   answerId: ID!
-  routingValue: RoutingConditionValueInput!
+  routingRule: ID!
 }
 
-input UpdateRoutingConditionInput {
+input UpdateRoutingConditionAnswerIdInput {
```
Comment (WickyNilliams) [inline index.js]: drop `Id` from mutation and input names

File: index.js [hunk 6]
```diff
@@ -391,47 +392,45 @@ input UpdateRoutingRuleSetInput {
...
 
+input UndeleteRoutingRuleInput {
+  id: ID!
+}
+
 input CreateRoutingConditionInput {
   comparator: RoutingComparator!
   answerId: ID!
-  routingValue: RoutingConditionValueInput!
+  routingRule: ID!
```
Comment (WickyNilliams) [inline index.js]: `routingRuleId`

File: index.js [hunk 7]
```diff
@@ -391,47 +392,45 @@ input UpdateRoutingRuleSetInput {
...
 
 input DeleteRoutingConditionInput {
   id: ID!
 }
 
-input RoutingConditionValueInput {
-  idArrayValue: [ID]
+input ToggleConditionOptionInput {
+  optionId: ID!
+  condition: ID!
```
Comment (WickyNilliams) [inline index.js]: `conditionId`

**Pull request merged**